tests/CommunityToolkit.Mvvm.UnitTests/Test_ObservablePropertyAttribute.cs

```diff
@@ -360,6 +361,105 @@ public void Test_OnPropertyChangingAndChangedPartialMethodWithAdditionalValidati
+
+        NullabilityInfo leftInfo = info.GenericTypeArguments[0];
+
+        Assert.AreEqual(typeof(Foo<string?, int>.Bar<object?>), leftInfo.Type);
+        Assert.AreEqual(NullabilityState.Nullable, leftInfo.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, leftInfo.WriteState);
+        Assert.AreEqual(3, leftInfo.GenericTypeArguments.Length);
+
+        NullabilityInfo leftInfo0 = leftInfo.GenericTypeArguments[0];
+
+        Assert.AreEqual(typeof(string), leftInfo0.Type);
+        Assert.AreEqual(NullabilityState.Nullable, leftInfo0.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, leftInfo0.WriteState);
+
+        NullabilityInfo leftInfo1 = leftInfo.GenericTypeArguments[1];
+
+        Assert.AreEqual(typeof(int), leftInfo1.Type);
+        Assert.AreEqual(NullabilityState.NotNull, leftInfo1.ReadState);
+        Assert.AreEqual(NullabilityState.NotNull, leftInfo1.WriteState);
+
+        NullabilityInfo leftInfo2 = leftInfo.GenericTypeArguments[2];
+
+        Assert.AreEqual(typeof(object), leftInfo2.Type);
+        Assert.AreEqual(NullabilityState.Nullable, leftInfo2.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, leftInfo2.WriteState);
+
+        NullabilityInfo rightInfo = info.GenericTypeArguments[1];
+
+        Assert.AreEqual(typeof(StrongBox<Foo<int, string?>.Bar<object>?>), rightInfo.Type);
+        Assert.AreEqual(NullabilityState.Nullable, rightInfo.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, rightInfo.WriteState);
+        Assert.AreEqual(1, rightInfo.GenericTypeArguments.Length);
+
+        NullabilityInfo rightInnerInfo = rightInfo.GenericTypeArguments[0];
+
+        Assert.AreEqual(typeof(Foo<int, string?>.Bar<object>), rightInnerInfo.Type);
+        Assert.AreEqual(NullabilityState.Nullable, rightInnerInfo.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, rightInnerInfo.WriteState);
+        Assert.AreEqual(3, rightInnerInfo.GenericTypeArguments.Length);
+
+        NullabilityInfo rightInfo0 = rightInnerInfo.GenericTypeArguments[0];
+
+        Assert.AreEqual(typeof(int), rightInfo0.Type);
+        Assert.AreEqual(NullabilityState.NotNull, rightInfo0.ReadState);
+        Assert.AreEqual(NullabilityState.NotNull, rightInfo0.WriteState);
+
+        NullabilityInfo rightInfo1 = rightInnerInfo.GenericTypeArguments[1];
+
+        Assert.AreEqual(typeof(string), rightInfo1.Type);
+        Assert.AreEqual(NullabilityState.Nullable, rightInfo1.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, rightInfo1.WriteState);
+
+        NullabilityInfo rightInfo2 = rightInnerInfo.GenericTypeArguments[2];
+
+        Assert.AreEqual(typeof(object), rightInfo2.Type);
+        //Assert.AreEqual(NullabilityState.NotNull, rightInfo2.ReadState);
+        //Assert.AreEqual(NullabilityState.NotNull, rightInfo2.WriteState);
+
+        // The commented out lines are to work around a weird behavior of the NullabilityInfo API there.
```

Comment:
I think you mentioned offline that there was an issue filed with the Rosyln team on this? Could you link it in the comment?

Reply:
Done in https://github.com/CommunityToolkit/dotnet/pull/160/commits/393a2232296dd90cb3cc96977f1728a88e52a8a4 👍